src/Fixer/PhpUnit/PhpUnitAttributesFixer.php

```diff
@@ -0,0 +1,482 @@
+            } else {
+                $tokens[$index] = new Token([T_DOC_COMMENT, $docBlock->getContent()]);
+            }
+        }
+    }
+
+    private function prepareFixingMap(): void
+    {
+        // annotations that map to attribute without parameters
+        foreach ([
+            'after',
+            'afterClass',
+            'before',
+            'beforeClass',
+            'coversNothing',
+            'doesNotPerformAssertions',
+            'large',
+            'medium',
+            'runInSeparateProcess',
+            'runTestsInSeparateProcesses',
+            'small',
+            'test',
+            'preCondition',
+            'postCondition',
+        ] as $annotation) {
+            $this->fixingMap[$annotation] = 'fixWithoutParameters';
+        }
+
+        // annotations that map to attribute with single string value
+        foreach (['group', 'testDox', 'ticket'] as $annotation) {
+            $this->fixingMap[$annotation] = 'fixWithSingleStringValue';
+        }
+
+        // annotations that map from 'enabled'/'disabled' value to attribute with boolean value
+        foreach (['preserveGlobalState', 'backupStaticAttributes', 'backupGlobals'] as $annotation) {
+            $this->fixingMap[$annotation] = 'fixWithEnabledDisabledValue';
+        }
+
+        // annotations that has custom mapping function
+        $this->fixingMap['covers'] = 'fixCovers';
+        $this->fixingMap['dataProvider'] = 'fixDataProvider';
+        $this->fixingMap['depends'] = 'fixDepends';
+        $this->fixingMap['requires'] = 'fixRequires';
+        $this->fixingMap['uses'] = 'fixUses';
+        $this->fixingMap['testWith'] = 'fixTestWith';
+    }
+
+    private static function shouldBeFixed(string $annotationName, string $annotationScope): bool
+    {
+        if (
+            'method' === $annotationScope
+            && \in_array($annotationName, ['covers', 'large', 'medium', 'runTestsInSeparateProcesses', 'small', 'uses'], true)
+        ) {
+            return false;
+        }
+
+        if (
+            'class' === $annotationScope
+            && \in_array($annotationName, ['after', 'afterClass', 'before', 'beforeClass', 'dataProvider', 'depends', 'postCondition', 'preCondition', 'runInSeparateProcess', 'test', 'testWith'], true)
+        ) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * @return list<Token>
+     */
+    private static function fixWithoutParameters(Tokens $tokens, int $index, Annotation $annotation): array
+    {
+        return self::createAttributeTokens($tokens, $index, self::getAttributeNameForAnnotation($annotation));
+    }
+
+    /**
+     * @return list<Token>
+     */
+    private static function fixWithSingleStringValue(Tokens $tokens, int $index, Annotation $annotation): array
+    {
+        Preg::match(
+            sprintf('/@%s\s+(.*\S)(?:\R|\s*\*+\\/$)/', $annotation->getTag()->getName()),
```

Comment:
- the regex would be good to be deduplicated
- link to a parsing phpunit source would be good so the parsing can be checked if 1:1 with native phpunit

Reply:
> * the regex would be good to be deduplicated

done

> * link to a parsing phpunit source would be good so the parsing can be checked if 1:1 with native phpunit

I'd rather not, for 2 reasons:
1. it is not a single place:
- https://github.com/sebastianbergmann/phpunit/blob/11.0.3/src/Metadata/Parser/AnnotationParser.php#L63
- https://github.com/sebastianbergmann/phpunit/blob/11.0.3/src/Metadata/Parser/Annotation/DocBlock.php#L41-L45
also, it was made differently in older PHPUnit: https://github.com/sebastianbergmann/phpunit/blob/9.6.16/src/Util/Annotation/DocBlock.php

2. let's focus on providing good tests, so they keep the fixer working properly